src/backend/expungeservice/endpoints/users.py

```diff
@@ -2,24 +2,50 @@
 from flask import request, abort, jsonify
 from werkzeug.security import generate_password_hash
 
-# TODO temp hack - replace with table from DB
-users = {}
+from flask import g
+from expungeservice.database.user import create_user, get_user_by_email
+from expungeservice.endpoints.auth import auth_required
+from psycopg2.errors import UniqueViolation
+
 
 class Users(MethodView):
+    @auth_required
     def post(self):
         data = request.get_json()
 
         if data == None:
             abort(400)
 
-        # TODO temp hack - replace with table from DB
-        if data['email'] in users:
-            return 'User already exists! Please login', 202
+        """
+        - the @auth_required decorator identifies the logged in user. If no auth, return 401 UNAUTHORIZED.
```

Comment:
Should this be a comment for the auth_required method? Better to have it there, I think, someone who is reading this method and isn't sure what `auth_required` would ideally look at the method definition.

Reply:
Sounds good, moving that over.